serving/src/main/java/feast/serving/specs/CachedSpecService.java

```diff
@@ -154,22 +177,50 @@ public void scheduledPopulateCache() {
                 ListFeatureSetsRequest.newBuilder()
                     .setFilter(
                         ListFeatureSetsRequest.Filter.newBuilder()
+                            .setProject(subscription.getProject())
                             .setFeatureSetName(subscription.getName())
                             .setFeatureSetVersion(subscription.getVersion()))
                     .build());
 
-        for (FeatureSet featureSet : featureSetsResponse.getFeatureSetsList()) {
-          FeatureSetSpec featureSetSpec = featureSet.getSpec();
-          featureSetSpecs.put(
-              String.format("%s:%s", featureSetSpec.getName(), featureSetSpec.getVersion()),
-              featureSetSpec);
+        for (FeatureSet fs : featureSetsResponse.getFeatureSetsList()) {
+          FeatureSetSpec spec = fs.getSpec();
+          featureSets.put(generateFeastId(spec.getProject(), spec.getName(), spec.getVersion()), spec);
         }
       } catch (StatusRuntimeException e) {
         throw new RuntimeException(
             String.format("Unable to retrieve specs matching subscription %s", subscription), e);
       }
     }
-    return featureSetSpecs;
+    return featureSets;
+  }
+
+  private Map<String, String> getFeatureToFeatureSetMapping(
+      Map<String, FeatureSetSpec> featureSets) {
+    HashMap<String, String> mapping = new HashMap<>();
+
+    featureSets.values().stream()
+        .collect(groupingBy(fs -> Triple.of(fs.getProject(), fs.getName(), fs.getVersion())))
+        .forEach(
+            (k, v) -> {
```

Comment:
Would make it a lot easier to see what is going on here if `k` and `v` had descriptive names